pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -193,4 +211,29 @@ protected void removeSegmentFromStore(String tableName, String segmentId) {
       LOGGER.info("localDiskDir is not configured, won't delete segment {} from disk", segmentId);
     }
   }
+
+  protected void removeSegmentsFromDeletedDirectory(int deleteRetention) {
```

Comment:
It would be better to separate the functionality into two, so code can be reused.
1. Identify what files need to be deleted.
2. Given an list of files, perform deletion.